pkg/receive/multitsdb.go

```diff
@@ -767,6 +771,7 @@ func (t *MultiTSDB) startTSDB(logger log.Logger, tenantID string, tenant *tenant
 		options = append(options, store.WithCuckooMetricNameStoreFilter())
 	}
 	tenant.set(store.NewTSDBStore(logger, s, component.Receive, lset, options...), s, ship, exemplars.NewTSDB(s, lset))
+	t.addTenantLocked(tenantID, tenant) // need to update the client list once store is ready & client != nil
```

Comment:
As far as I can tell, this is the "meat" of your fix i.e. this was missing, right?

Reply:
Yes, this line is necessary because a unit test would fail, the reason is when a tenant is added, it could have nil TSDBClient due to not set TSDB store yet.

The original implementation would be fine since `tenant.client()` is called every time so it would create clients later. However after this fix, we need to make sure clients are up-to-date only when a tenant is added or pruned.